transformations/transformations/scc_cuf.py

```diff
@@ -60,6 +61,35 @@ def driver_variable_declaration(self, routine, variables):
                 routine.body.append(ir.Deallocation((var.clone(dimensions=None),)))
 
 
+class HoistTemporaryArraysPragmaOffloadTransformation(HoistVariablesTransformation):
+    """
+    Synthesis part for variable/array hoisting, offload via pragmas e.g., OpenACC.
+    """
+
+    def driver_variable_declaration(self, routine, variables):
+        """
+        Standard Variable/Array declaration including
+        device offload via pragmas.
+
+        Parameters
+        ----------
+        routine: :any:`Subroutine`
+            The subroutine to add the variable declaration
+        var: :any:`Variable`
+            The variable to be declared
+        """
+        for var in variables:
+            routine.variables += tuple([var.clone(scope=routine)])
```

Comment:
Note that editing the variables list multiple times can be very costly, because it will compile the list of variables from scratch from the declarations on every invocation, and then check and insert declarations. I think it would be much faster to do it like this:
```suggestion
        routine.variables += tuple(var.clone(scope=routine) for var in variables)
```